server/service/src/invoice_line/prescription_line/update/update.rs

```diff
@@ -0,0 +1,391 @@
+use crate::{
+    invoice_line::{
+        common_update_line::{generate, validate, UpdateInvoiceLine, UpdateInvoiceLineError},
+        query::get_invoice_line,
+    },
+    service_provider::ServiceContext,
+};
+use repository::{InvoiceLine, InvoiceLineRowRepository, InvoiceRowType, StockLineRowRepository};
+
+type OutError = UpdateInvoiceLineError;
+
+pub fn update_prescription_line(
+    ctx: &ServiceContext,
+    input: UpdateInvoiceLine,
+) -> Result<InvoiceLine, OutError> {
+    let updated_line = ctx
+        .connection
+        .transaction_sync(|connection| {
+            let (line, item, batch_pair, invoice) = validate(
+                &input,
+                &ctx.store_id,
+                &connection,
+                InvoiceRowType::Prescription,
+            )?;
+
+            let (update_line, batch_pair) = generate(input, line, item, batch_pair, invoice)?;
+            InvoiceLineRowRepository::new(&connection).upsert_one(&update_line)?;
+
+            let stock_line_repo = StockLineRowRepository::new(&connection);
+            stock_line_repo.upsert_one(&batch_pair.main_batch)?;
+            if let Some(previous_batch) = batch_pair.previous_batch_option {
+                stock_line_repo.upsert_one(&previous_batch)?;
+            }
```

Comment:
The only difference to update_outbound_shipment_line() is the InvoiceRowType, right? Would it have been easier to just have this as an parameter to update_outbound_shipment_line()? then you don't have to move things to common_update_line...

Reply:
I was going to refactor to have it all in one common service like with the insert line... 

> As per our conversation, it's probably less logic for now to ignore placeholder/unallocated line in UI rather then adding unallocated line logic in backed end (including when updating prescription)
As for Andrei's comment, this belongs in the Update Prescription PR, not the Update Prescription Line PR XD